shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -309,7 +324,7 @@ - (void)setMarkedText:(NSString*)markedText selectedRange:(NSRange)markedSelecte
 
   NSUInteger selectionLocation = markedSelectedRange.location + markedTextRange.location;
   selectedRange = NSMakeRange(selectionLocation, markedSelectedRange.length);
-  [self setSelectedTextRange:[FlutterTextRange rangeWithNSRange:selectedRange]
+  [self setSelectedTextRange:[FlutterTextRange rangeWithNSRange:selectedRange withText:self.text]
```

Comment:
Isn't `setSelectedTextRange` already doing this check (twice now, maybe)?

Reply:
It's `setTextInputState` vs `setSelectedTextRange`, but I'll just extra the filtering into a static.